### PR TITLE
PARQUET-629: RowGroupSerializer should only close itself once

### PR DESCRIPTION
**Pull request by xhochy**

### CODE REVIEW

File: src/parquet/file/writer-internal.cc [hunk 1]
```diff
@@ -148,16 +148,19 @@ ColumnWriter* RowGroupSerializer::NextColumn() {
 }
 
 void RowGroupSerializer::Close() {
-  if (current_column_index_ != schema_->num_columns() - 1) {
-    throw ParquetException("Not all column were written in the current rowgroup.");
+  if (!closed_) {
```
Comment (wesm) [inline src/parquet/file/writer-internal.cc]: Perhaps we should have a DCHECK(!closed_), unless there's a reason we want this to be idempotent?

Comment (xhochy) [inline src/parquet/file/writer-internal.cc]: I'd prefer to have close idempotent. Thus you can call it explicitly in the code or implicitly in the destructor.

Comment (wesm) [inline src/parquet/file/writer-internal.cc]: Works for em

**Comment by wesm** [comment]
+1, thank you

**Pull request closed without merging**